### ISSUE TITLE
Compatibility with other react-native modules

### ISSUE THREAD

**Issue by pedro** [wix/react-native-controllers#40]
One of the challenges migrating to react-native-controllers is that a lot of the other extensions stop working, because they assume the root controller never changes (and is the top one in the window hierarchy).

To fix I had to change every single dependency to lazily fetch the root controller, and call `presentedViewController` on it, like:

``` obj-c
-(UIViewController *)getRootController
{
  id<UIApplicationDelegate> appDelegate = [UIApplication sharedApplication].delegate;
  return appDelegate.window.rootViewController.presentedViewController;
}
```

Is there any way we can fix this more broadly from react-native-controllers itself? Or any suggestions of upstream changes that would allow this to coexist with other extensions?

Thanks!

**Comment by talkol** [comment]
Can you give some examples of extensions that stopped working?

**Comment by drytdubs** [comment]
I ran into this issue when trying to combine react-native-maps with react-native-navigation on a small side project. Admittedly, I don't have a ton of experience in fixing the native level linkings, so I was unable to get a fix myself in the time I had to work on it.

I was working on the Android side linking issue and tried following both the "manual" linking instructions and running the react-native link command.
